### ISSUE TITLE
Overlaped items when scrolling left right

### ISSUE THREAD

**Issue by RG72** [almende/vis#1982]
Overlap items
![visjsoverlap](https://cloud.githubusercontent.com/assets/1561460/17048479/10f852b4-5008-11e6-93ce-6b0cbe14e13b.png)

After zoom in and zoom out. It's OK.
![visjsafterzoominout](https://cloud.githubusercontent.com/assets/1561460/17048493/242609da-5008-11e6-8592-7964c0768f65.png)

Example at http://jsbin.com/cilidonuno/1/edit?html,output

**Comment by mojoaxel** [comment]
The problem seams to occure when "new" items get rendered while moving the timeline (pan).

**Comment by sk29110** [comment]
FYI I only have 2 items in the timeline view still able to reproduce the problem. Just you like mentioned when you scroll(pan) so that 2 items are not in the visible view and scroll back to show those two items they are collapsed into each other.

![2016-11-24_11-16-37](https://cloud.githubusercontent.com/assets/626182/20608554/c46711d2-b237-11e6-8b69-a532b3cf107b.png)

**Comment by bradh** [comment]
I have an idea why this might be occurring. In ItemSet `redraw()`, we mark the `stackDirty` with this logic:
```
  // check whether zoomed (in that case we need to re-stack everything)
  // TODO: would be nicer to get this as a trigger from Range
  var visibleInterval = range.end - range.start;
  var zoomed = (visibleInterval != this.lastVisibleInterval) || (this.props.width != this.props.lastWidth);
  if (zoomed) this.stackDirty = true;
  this.lastVisibleInterval = visibleInterval;
  this.props.lastWidth = this.props.width;

  var restack = this.stackDirty;
```

I tried forcing the restack (set `this.stackDirty` to `true`). I could not reproduce the problem with that in place.

So I think we need to consider if some additional logic is needed here. Perhaps keep old start date (and perhaps end date) and if that has changed then restack.

Fancier logic would check if the change actually needs to restack (e.g. by checking for intersection of start / end dates on each item visible), but that might be more work than is useful.

**Comment by yotamberk** [comment]
@bradh nice investigation work! Do you know to implement the new added logic needed?

**Comment by bradh** [comment]
I'm still building up knowledge on ItemSet (which is a lot of code!), and have no clue how to automate the tests for this case yet. Time is limited (for everyone), so I'm assuming it will take me a while to get this done. Its an annoying case (two separate bugs, at least), so if you want to deal with it, that would be good. Otherwise I'll try to take a look.

**Comment by bradh** [comment]
Add a PR, for the simplest logic I could come up with. This could well be a place for some future refactoring.

**Comment by yotamberk** [comment]
fixed with #2774. Thanks @bradh !
